### PR TITLE
docs: HAProxy API Gateway

### PR DESCRIPTION
**Pull request by sixhobbits**

### CODE REVIEW

File: site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+-H 'Authorization: Bearer <TOKEN>'
+----
+
+Here, `TOKEN` is the value of the JWT that you copied from the `/api/login` call.
+
+This command should return `HTTP/1.1 200 OK`, indicating successful authorization to the HAProxy service.
+
+== Troubleshooting
+
+* The JWT issued by the FusionAuth Login API has an expiry. If you wait too long before using it to call the HAProxy service, the token will expire and the call will fail. You can resolve this by rerunning the curl command to obtain another JWT token and use the new token.
```
Comment (vcampitelli) [inline site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc:175]:
```suggestion
* The JWT issued by the FusionAuth Login API has an expiration date. If you wait too long before using it to call the HAProxy service, the token will expire and the call will fail. You can resolve this by rerunning the curl command to obtain another JWT token and use the new token.
```

Just a suggestion

File: site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+== Prerequisites
+
+* A FusionAuth instance running on a publicly accessible URL. You can spin up a link:/pricing[basic FusionAuth Cloud instance] or link:/docs/v1/tech/installation-guide/[install it on any server].
+* An HAProxy instance routed to an HTTP service. You can follow https://www.haproxy.com/blog/how-to-run-haproxy-with-docker/[this guide] to get a simple example running on Docker.
+
+[NOTE.note]
+====
+When creating your HAProxy container, be sure to use `haproxytech/haproxy-alpine:latest`, not `haproxytech/haproxy-alpine:2.4` as the guide instructs. Version 2.4 does not support the full set of JWT features needed in this tutorial.
+
+If you have trouble getting the `haproxy` container from the guide to start, make sure your `haproxy.cfg` file ends with a newline. Even whitespace on the last line will prevent the container from running.
```
Comment (vcampitelli) [inline site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc:25]:
```suggestion
If you have trouble getting the `haproxy` container from the guide to start, make sure your `haproxy.cfg` file ends with a newline. Even a whitespace on the last line will prevent the container from running.
```

Another suggestion

File: site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+image::api-gateways/HAProxy-Api-Gateway/haproxy-application-jwt-settings.png[Application JWT settings,width=1200px]
+
+Click the [uielement]#Save# button.
+
+After saving the new application, find it in the list of applications and click the [uielement]#View# button next to it (look for the green magnifying glass). Here you will find the application [field]#Id#. Record this value, as you will need it further on. It will be referred to later as `<YOUR_APPLICATION_ID>`.
+
+image::api-gateways/HAProxy-Api-Gateway/haproxy-application-id.png[Application id,width=1200px,role=bottom-cropped]
+
+Now, navigate to [breadcrumb]#Settings -> Key Master#. You will see the access key that you've just created in this list. By default, it has the name `Access token signing key generated for application <NAME_OF_APPLICATION>`. View the key and copy the entire contents of the [field]#PEM encoded# field under the [breadcrumb]#Public Key# section into a file in the same directory as you placed the `haproxy.cfg` file whilst setting up HAProxy in the quickstart guide. Name the file `pubkey.pem`. Remember this filename, as you will supply it to the HAProxy configuration file later.
```
Comment (vcampitelli) [inline site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc:49]: [nitpick (non-blocking):](https://conventionalcomments.org/) I think we could split _"View the key and ... in the quickstart guide"_ in two to make them simpler.

**Review by vcampitelli — CHANGES_REQUESTED** [review]
I think this looks good! I just left three simple suggestions and one change in the `Next Steps` links.

File: site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc (hunk 3 above)
Comment (sixhobbits) [reply site/docs/v1/tech/developer-guide/api-gateways/haproxy-api-gateway.adoc]: done

**Comment by sixhobbits** [comment]
@vcampitelli - we've committed your suggestions and adjusted the text in the place you requested.

**Review by vcampitelli — APPROVED** [review]

File: site/assets/img/docs/api-gateways/HAProxy-Api-Gateway/haproxy-api-key-permissions.png
Comment (vcampitelli) [inline site/assets/img/docs/api-gateways/HAProxy-Api-Gateway/haproxy-api-key-permissions.png:1]: @sixhobbits Actually, I just saw that these screenshots are from FusionAuth 1.42.1 and we are currently on 1.45.2. Can you please update your instance and take the screenshots again? Sorry that I missed it earlier.

**Review by vcampitelli — CHANGES_REQUESTED** [review]

File: site/assets/img/docs/api-gateways/HAProxy-Api-Gateway/haproxy-api-key-permissions.png
Comment (rideam) [reply site/assets/img/docs/api-gateways/HAProxy-Api-Gateway/haproxy-api-key-permissions.png:1]: @vcampitelli I have updated the images

**Comment by mooreds** [comment]
@vcampitelli Can you take another look at this and let me know if it is ready to go?

**Comment by mooreds** [comment]
@vcampitelli ?

**Review by vcampitelli — APPROVED** [review]

**Comment by mooreds** [comment]
Closed under https://github.com/FusionAuth/fusionauth-site/pull/2308

**Pull request closed without merging**